lib/dateTime.js

```diff
@@ -0,0 +1,578 @@
+	if (settings.overrides.date) {
+		merge(descriptor, settings.overrides.date);
+	}
+
+	return descriptor;
+
+}
+
+export function formatTime(date, options) {
+
+	const descriptor = getDateTimeDescriptor();
+	const settings = getDocumentLocaleSettings();
+
+	options = options || {};
+
+	const timezone = options.timezone || settings.timezone.name;
+	const format = descriptor.formats.timeFormats[options.format]
+		|| options.format || descriptor.formats.timeFormats['short'];
+
+	const hour = date.getHours();
+	let hour12 = hour % 12;
+	if (hour12 === 0) {
+		hour12 = 12;
+	}
+
+	const replacements = {
+		'HH': prePadByZero(date.getHours(), 2),
+		'H': date.getHours().toString(),
+		'hh': prePadByZero(hour12, 2),
+		'h': hour12,
+		'mm': prePadByZero(date.getMinutes(), 2),
+		'tt': (hour > 11) ? descriptor.calendar.dayPeriods.pm : descriptor.calendar.dayPeriods.am,
+		'ZZZ': timezone
+	};
+
+	const value = processPattern(format, replacements);
+	return value;
+
+}
+
+export function parseTime(input, options) {
+
+	if (input === undefined || input === null || input === '') {
+		return null;
+	}
+
+	const descriptor = getDateTimeDescriptor();
+
+	const reDigits = new RegExp('(\\d+)', 'g');
+	const match = input.match(reDigits);
+	if (match === null) {
+		return null;
+	}
+
+	options = options || {};
+	const nowProvider = options.nowProvider || function() { return new Date(); };
+	const reAm = buildDayPeriodRe(descriptor.calendar.dayPeriods.am);
+	const rePm = buildDayPeriodRe(descriptor.calendar.dayPeriods.pm);
+	const today = nowProvider();
+	const isMorning = (today.getHours() < 12);
+	const digits = match.join('');
+	const leadingZero = (digits.substr(0, 1) === '0');
+
+	let hour = 0;
+	let minute = 0;
+	switch (digits.length) {
+		case 1:
+			hour = digits.substr(0, 1);
+			break;
+		case 2:
+			hour = digits.substr(0, 2);
+			break;
+		case 3:
+			hour = digits.substr(0, 1);
+			minute = digits.substr(1, 2);
+			break;
+		default:
+			hour = parseInt(digits.substr(0, 2));
+			minute = parseInt(digits.substr(2, 2));
+			break;
+	}
+
+	hour = Math.min(Math.max(parseInt(hour, 10), 0), 23);
+	minute = Math.min(Math.max(parseInt(minute, 10), 0), 59);
+
+	if (!descriptor.hour24 && hour < 13) {
+
+		const matchPm = input.match(rePm);
+		const matchAm = input.match(reAm);
+		const noAmPm = (matchAm === null && matchPm === null);
+
+		if (matchPm !== null || (noAmPm && !isMorning && !leadingZero)) {
+			hour += 12;
+			if (hour === 24) {
+				hour = 12;
+			}
+		} else if (hour === 12) {
+			hour = 0;
+		}
+
+	}
+
+	const time = new Date(today.getFullYear(), today.getMonth(), today.getDate(), hour, minute, 0);
+	return time;
+
+}
+
+export function formatDate(date, options) {
+
+	const descriptor = getDateTimeDescriptor();
+
+	options = options || {};
+	options.format = options.format || 'short';
+
+	let format = descriptor.formats.dateFormats[options.format];
+	if (format === undefined) {
+		format = options.format;
+	}
+
+	switch (format) {
```

Comment:
This confused me a bit at first, but I think the idea is that these are other dateFormats that are consistent in their format and don't depend on the locale. Would it make sense to move these to https://github.com/Brightspace/intl/blob/58d25f5ce1ef3b62bb71c3756fd18fd11e639155/lib/dateTime.js#L333 so they're all together? I'm fine with it as is, but it would just put them all in one place (though maybe there's something I'm missing and these are way better here)

Reply:
I like that idea -- will do. Originally yeah it was handled separately since they're not "standard" formats... but now that everything isn't broken out across many files I think just moving it up to the descriptor should be fine.